frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx

```diff
@@ -61,23 +62,55 @@ function JsonConfigField(props: {
     templating?: boolean
 }): JSX.Element {
     const { globalsWithInputs } = useValues(hogFunctionConfigurationLogic)
+    const [key] = useState(`json_field_${Math.random().toString(36).substring(2, 11)}`)
+
+    // Set up validation logic for this JSON field
+    const logic = hogFunctionInputLogic({
+        fieldKey: key,
+        initialValue:
+            props.value === undefined || props.value === null
+                ? '{}'
+                : typeof props.value !== 'string'
+                ? JSON.stringify(props.value, null, 2)
+                : props.value,
+        onChange: props.onChange,
+    })
+
+    const { error } = useValues(logic)
+    const { setJsonValue } = useActions(logic)
+
+    // Format initial value for display
+    const initialValue =
```

Comment:
Same with this. The value should come from the logic if thats how you are doing this